ui/hooks/useTokenFiatAmount.js

```diff
@@ -42,15 +42,16 @@ export function useTokenFiatAmount(
     shallowEqual,
   );
 
-  const contractMarketData = chainId
-    ? Object.entries(allMarketData[chainId]).reduce(
-        (acc, [address, marketData]) => {
-          acc[address] = marketData?.price ?? null;
-          return acc;
-        },
-        {},
-      )
-    : null;
+  const contractMarketData =
+    chainId && allMarketData[chainId]
```

Comment:
After the user removed the network and added it back while having the setting  `Show balance and token price checker` OFF, the `allMarketData[chainId] `for the chainId was `undefined`, making `Object.entries(allMarketData[chainId])` throw the error we see in the issue `Cannot convert undefined or null to object `
Once the user turns the setting back on, it will populate the market data with the values again.